### PR TITLE
Verify signmsg

### PR DESCRIPTION
**Pull request by benma**
Motivation: I saw that Bity makes you sign this kind of message when buying:

```
This message is part of the process of placing an exchange order on bity.com. To proceed, Bity is required to verify that you own the destination crypto address. By signing this message with your wallet, you are confirming 1) that you are the sole owner of the crypto address below and 2) that you will be sending your own funds to Bity. If you did not initiate this process yourself, do not sign this message.

Your order id: 69b87653-.....
Your IBAN: CH1.........
Your crypto address: 36..........
Your crypto address type: BTC
```

The newlines make this show as a hex string. After this PR a message like this can be easily verified.

To test, enter this in send_message for 21) (sign eth msg):

```
This message is part of the process of placing an exchange order on bity.com. To proceed, Bity is required to verify that you own the destination crypto address. By signing this message with your wallet, you are confirming 1) that you are the sole owner of the crypto address below and 2) that you will be sending your own funds to Bity. If you did not initiate this process yourself, do not sign this message.\n\nYour order id: 69b876.......d9f\nYour IBAN: CH17...0\nYour crypto address: 3659........T\nYour crypto address type: BTC
```

### CODE REVIEW

**Review by x1ddos — COMMENTED** [review]
This is cool but how do I sign it on a btc-only firmware?

File: src/rust/bitbox02-rust/src/workflow/verify_message.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+extern crate alloc;
+use alloc::vec::Vec;
+
+use super::confirm;
+
+/// Verify a message to be signed.
+///
+/// If the bytes are all printable ascii chars, the message is
+/// confirmed one line at a time.
```
[the remark this answers is not in the document]
Comment (benma) [reply src/rust/bitbox02-rust/src/workflow/verify_message.rs:23]: it say `one line at a time`. Isn't that clear?

File: src/rust/util/src/ascii.rs [hunk 2]
```diff
@@ -18,10 +18,13 @@
 /// !"#$%&\'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~
 /// ```
 ///
-/// Unlike std's is_ascii, control characters such as newline, tab, etc. are not part
-/// of the set.
-pub fn is_printable_ascii<T: AsRef<[u8]>>(bytes: T) -> bool {
-    bytes.as_ref().iter().all(|&b| b >= 32 && b <= 126)
+/// Note that newline, tab, etc. are not part of this set.
+/// If `allow_newline` is true, '\n' is also accepted.
+pub fn is_printable_ascii<T: AsRef<[u8]>>(bytes: T, allow_newline: bool) -> bool {
```
[the remark this answers is not in the document]
Comment (benma) [reply src/rust/util/src/ascii.rs]: Good idea, done.

File: src/rust/bitbox02-rust/src/workflow/verify_message.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (benma) [reply src/rust/bitbox02-rust/src/workflow/verify_message.rs]: extended the comment

**Review by x1ddos — APPROVED** [review]
LGTM

**Pull request merged**